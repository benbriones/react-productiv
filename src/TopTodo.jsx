import React from "react";

import Todo from "./Todo";

/** Shows the top todo.
 *
 * Props:
 * - todos
 *
 * TodoApp -> TopTodo
 */

// TODO: add, if no top, return null

function TopTodo({todos}) {
  // lowest-priority # is the highest priority

  let top = todos.reduce(
      (acc, cur) => cur.priority < acc.priority ? cur : acc, todos[0]);

  const { id, title, description, priority } = top;

  return <Todo id={id} title={title} description={description} priority={priority}/>;
}

export default TopTodo;