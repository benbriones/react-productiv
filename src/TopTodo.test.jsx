import {describe, it, expect } from "vitest";

import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";

describe("TopTodo", function () {
    const testTodos = [
        { id:4, title: "celebrate bday", description: "for dinner", priority:1 },
        { id:3, title: "eat carbonara", description:"buca di peppo", priority:2 },
        { id:2, title: "mow the lawn", description:"garden things", priority:3 }
    ]

  it("renders without crashing", function () {
    render(<TopTodo todos={testTodos} />);
  });

  it("matches snapshot", function () {
    const { container } = render(<TopTodo todos={testTodos} />)
    expect(container).toMatchSnapshot();
  });

  it ("rendered contains expected", function () {
    const { container } = render(<TopTodo todos={testTodos} />)
    const topTodo = container.querySelector(".Todo")
    expect(topTodo).toContainHTML("celebrate");
  });

  it ("did not render if not topTodo", function () {
    const { container } = render(<TopTodo todos={testTodos} />)
    const topTodo = container.querySelector(".Todo")
    expect(topTodo).not.toContainHTML("buca di peppo");
  });
});

