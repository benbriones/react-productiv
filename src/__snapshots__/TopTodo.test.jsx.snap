// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TopTodo > matches snapshot 1`] = `
<div>
  <div
    class="Todo"
    id="4"
  >
    <div>
      <b>
        celebrate bday
      </b>
       
      <small>
        (Priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        for dinner
      </small>
    </div>
  </div>
</div>
`;
